src/join.rs

```diff
@@ -66,6 +66,75 @@ bitset_and!{A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P}
 /// The purpose of the `Join` trait is to provide a way
 /// to access multiple storages at the same time with
 /// the merged bit set.
+///
+/// Joining component storages means that you'll only get values where
+/// for a given entity every storage has an associated component.
```

Comment:
Should we draw parallel to SQL? Or is that just confusing?

Reply:
I don't find it very useful tbh